### PR TITLE
Support copying the location of a bundle or feature to the clipboard

### PR DESCRIPTION
**Pull request by merks**
https://github.com/eclipse-pde/eclipse.pde/issues/676

### CODE REVIEW

**Comment by merks** [comment]
@HannesWell 

FYI, I squash and merge this before the end of the day, unless you cry out before then...

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/CopyLocationAction.java
Comment (HannesWell) [inline ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/CopyLocationAction.java:42]: Is there a specific reason why you used `var` here and below instead of the actual type?
Personally I would have just used the type, since it isn't a complex generics chain here and I would say that it is easier to understand with just the type written out.

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/TargetContentsGroup.java
Comment (HannesWell) [inline ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/TargetContentsGroup.java:321]: Nice catch.

**Review by HannesWell — APPROVED** [review]
This is is nice. Looks good and works well. 👍🏽

It would be great to have somehing similar for the Target Platform State view, where the actual bundle's location is only displayed but one cannot copy it.
For PDE's Plug-in Dependencies Classpath Container its the same.

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/CopyLocationAction.java
Comment (merks) [reply ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/CopyLocationAction.java:42]: Oh, oh.  We need a var-policy!  To use or when to use, that is the question?  😶

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/TargetContentsGroup.java
Comment (merks) [reply ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/TargetContentsGroup.java:321]: That took much longer to figure out than I care to admit.

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/CopyLocationAction.java
Comment (merks) [reply ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/CopyLocationAction.java:42]: I will change it if you wish...

Comment (HannesWell) [reply ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/CopyLocationAction.java:42]: > Oh, oh. We need a var-policy! To use or when to use, that is the question? 😶

That would be good, yes. 😄
Personally I only use var when the type is too 'long' (e.g. a complex, nested generic type) or something internal that generates a discouraged-access warning. Then one can often use the fully-qualified-name in the constructor and `var` for the type and suppress the warning for the variable and often get that in one line:
`var name = new org.eclipse.foo.bar.internal.ClassName()`
The downside of var is that one cannot read the type from the variable declaration anymore. If one calls a constructor on the RHS of the assignment it is still simple to figure out, but the return type of methods is often hard/impossible to guess.
So I only use war in the cases above or if the type is really, really uninteresting.

But that's just my personal preference, but I think it would be good to have a general agreement about the usage of var in the Eclipse SDK. Since we don't have that now, I would leave that to your discretion since it is your change :)

**Comment by merks** [comment]
> This is is nice. Looks good and works well. 👍🏽
> 
> It would be great to have somehing similar for the Target Platform State view, where the actual bundle's location is only displayed but one cannot copy it. For PDE's Plug-in Dependencies Classpath Container its the same.

This latest update reuses the action for the Platform State view too.

**Pull request merged**
